addon/templates/components/course-materials.hbs

```diff
@@ -1,106 +1,153 @@
 <div class="material-list">
-  <h3>{{t "general.courseLearningMaterials"}} ({{get (await course.learningMaterials) "length"}})</h3>
-  {{#if (gt (get (await course.learningMaterials) "length") 0)}}
-    <table>
-      <thead>
+  <h3>
+    {{t "general.courseLearningMaterials"}} ({{get (await this.course.learningMaterials) "length"}})
+  </h3>
+
+  <span class="filter-course-lms">
+    <input
+      placeholder={{t "general.filterPlaceholder"}}
+      value={{this.courseQuery}}
+      oninput={{perform this.setCourseQuery value="target.value"}}>
+  </span>
+
+  <table data-test-course-table>
+    <thead>
+      <tr>
+        {{#sortable-th
+          colspan=4
+          sortedAscending=this.clmSortedAscending
+          sortedBy=(or (eq @courseSort "title") (eq @courseSort "title:desc"))
+          click=(action "courseSortBy" "title")}}
+          {{t "general.title"}}
+        {{/sortable-th}}
+        {{#sortable-th
+          sortedAscending=this.clmSortedAscending
+          sortedBy=(or (eq @courseSort "type") (eq @courseSort "type:desc"))
+          click=(action "courseSortBy" "type")}}
+          {{t "general.type"}}
+        {{/sortable-th}}
+        {{#sortable-th
+          sortedAscending=this.clmSortedAscending
+          sortedBy=(or (eq @courseSort "author") (eq @courseSort "author:desc"))
+          click=(action "courseSortBy" "author")}}
+          {{t "general.author"}}
+        {{/sortable-th}}
+      </tr>
+    </thead>
+
+    <tbody>
+      {{#if (is-pending this.courseLearningMaterialObjects)}}
         <tr>
-          <th colspan="4">{{t "general.title"}}</th>
-          <th>{{t "general.type"}}</th>
-          <th>{{t "general.author"}}</th>
+          <td colspan="6" align="center">
+            {{fa-icon "spinner" class="orange" size="2x" spin=true}}
+          </td>
         </tr>
-      </thead>
-      <tbody>
-        {{#each (await courseLearningMaterialObjects) as |lmObject|}}
+      {{else}}
+        {{#each (sort-by @courseSort (await this.filteredCourseLearningMaterialObjects))
+          as |lmObject|}}
           <tr>
             <td colspan="4">
-              {{#if (contains lmObject.type typesWithUrl)}}
-                <a href={{lmObject.url}} target="_blank" rel="noopener">{{lmObject.title}}</a>
+              {{#if (contains lmObject.type this.typesWithUrl)}}
+                <a href={{lmObject.url}} rel="noopener" target="_blank" >
+                  {{lmObject.title}}
+                </a>
               {{else}}
                 {{lmObject.title}}<br>
                 <small>{{lmObject.citation}}</small>
               {{/if}}
             </td>
-            <td>{{lmObject.type}}</td>
+            <td>{{capitalize lmObject.type}}</td>
             <td>{{lmObject.author}}</td>
           </tr>
+        {{else}}
+          <tr>
+            <td colspan="6" align="center">
+              {{if this.courseQuery "No Results Found" "No Course Learning Materials Available"}}
```

Comment:
Please pull this verbiage out and into translations.

Reply:
Fixed.